### PR TITLE
Add support for login with different emails

### PR DESCRIPTION
**Pull request by abuzar-deriv**
1. Added function setLoginUser: To switch user based on test vs production environment

2.Updated c_login() and getOAuthUrl(): getOAuthUrl() will now receive loginEmail & loginPassword as parameters instead of switching inside the function

3.Updated function calls of c_login() within tests where needed (The change is just on lines where c_login() is being called, rest are auto-formatting changes 😿 )

4.Added new env variables

### CODE REVIEW

**Review by markodonnell-deriv — COMMENTED** [review]
Quite a large PR and looks good in principle.

File: cypress/support/common.js [hunk 1]
```diff
@@ -1,105 +1,100 @@
...
+    cy.log('<solution>' + solution)
     cy.request({
       method: 'POST',
-      url: 'https://' + Cypress.env('configServer') + '/oauth2/api/v1/verify',
+      url:
+        'https://' + Cypress.env('configServer') + '/oauth2/api/v1/authorize',
       headers: {
-        'Origin': 'https://oauth.deriv.com',
+        Origin: 'https://oauth.deriv.com',
         'Content-Type': 'application/json',
```
Comment (bilalmajid-deriv) [inline cypress/support/common.js:28]: Should the '' here be removed?

File: cypress/support/common.js [hunk 2]
```diff
@@ -1,105 +1,100 @@
...
+      const bearerToken = response.body.token
+      cy.log('<bearer token>' + bearerToken)
+      expect(response.status).to.eq(200)
+
       cy.request({
         method: 'POST',
-        url: 'https://' + Cypress.env('configServer') + '/oauth2/api/v1/authorize',
+        url: 'https://' + Cypress.env('configServer') + '/oauth2/api/v1/login',
         headers: {
-          'Origin': 'https://oauth.deriv.com',
```
Comment (bilalmajid-deriv) [inline cypress/support/common.js:24]: same as above

File: cypress/support/common.js [hunk 3]
```diff
@@ -1,105 +1,100 @@
...
-    url: 'https://' + Cypress.env('configServer') + '/oauth2/authorize?app_id=' + Cypress.env('configAppId') + '&l=en&brand=deriv&date_first_contact=',
+    url:
+      'https://' +
+      Cypress.env('configServer') +
+      '/oauth2/authorize?app_id=' +
+      Cypress.env('configAppId') +
+      '&l=en&brand=deriv&date_first_contact=',
     headers: {
       'Content-Type': 'application/x-www-form-urlencoded',
-      'Origin': 'https://oauth.deriv.com'
```
Comment (bilalmajid-deriv) [inline cypress/support/common.js:86]: same as above

**Review by bilalmajid-deriv — COMMENTED** [review]
PR seems to be looking good 

but a question left for removing '' .

**Review by bilalmajid-deriv — CHANGES_REQUESTED** [review]
a new command for getWalletOAuthUrl() is added can you please pass loing email and login password for that as well accordingly.

**Comment by abuzar-deriv** [comment]
Good catch about quotes. It was prettier's doing; I've added a rule there now. As for `getWalletOAuthUrl()`, I don't think it's actually needed. I believe it was just added to handle different login credentials. I'll discuss with @afsaneh-rezaeinejad-deriv and remove it in another PR. Or if it's really required, then I'll add the login part there as well.

**Comment by bilalmajid-deriv** [comment]
> Good catch about quotes. It was prettier's doing; I've added a rule there now. As for `getWalletOAuthUrl()`, I don't think it's actually needed. I believe it was just added to handle different login credentials. I'll discuss with @afsaneh-rezaeinejad-deriv and remove it in another PR. Or if it's really required, then I'll add the login part there as well.

okay sure

**Review by meeyun-deriv — APPROVED** [review]
LGTM

**Review by markodonnell-deriv — APPROVED** [review]
Look good

**Review by bilalmajid-deriv — APPROVED** [review]
LGTM

**Pull request merged**